base/FileMetadataUtilities.F90

```diff
@@ -46,6 +51,110 @@ subroutine create(this,metadata,fname)
       this%filename = fName
    end subroutine create
 
+   function var_get_missing_value(this,var_name,rc) result(missing_value)
+      class(FileMetadataUtils), intent(inout) :: this
+      character(len=*), intent(in) :: var_name
+      integer, optional, intent(out) :: rc
+
+      real(REAL32) :: missing_value
+      integer :: status
+      type(Variable), pointer :: var
+
+      var => this%get_variable(var_name,_RC)
+      _ASSERT(associated(var),"no variable named "//var_name//" in file")
+      ! check _FillValue, we could do more, not sure what to do here like also check for missing_value ...
+      if (this%var_has_attr(var_name,"_FillValue")) then
+         missing_value = this%get_var_attr_real32(var_name,"_FillValue",_RC)
+      end if
+
+      _RETURN(_SUCCESS)
+   end function var_get_missing_value
+
+   logical function var_has_missing_value(this,var_name,rc)
+      class(FileMetadataUtils), intent(inout) :: this
+      character(len=*), intent(in) :: var_name
+      integer, optional, intent(out) :: rc
+     
+      integer :: status
+      type(Variable), pointer :: var
+
+      var => this%get_variable(var_name,_RC)
+      _ASSERT(associated(var),"no variable named "//var_name//" in file")
+      var_has_missing_value = var%is_attribute_present("_FillValue")
+
+      _RETURN(_SUCCESS)
+   end function var_has_missing_value
+
+   logical function var_has_attr(this,var_name,attr_name,rc)
+      class(FileMetadataUtils), intent(inout) :: this
+      character(len=*), intent(in) :: var_name
+      character(len=*), intent(in) :: attr_name
+      integer, optional, intent(out) :: rc
+     
+      integer :: status
+      type(Variable), pointer :: var
+
+      var => this%get_variable(var_name,_RC)
+      _ASSERT(associated(var),"no variable named "//var_name//" in file")
+      var_has_attr = var%is_attribute_present(attr_name)
+      _RETURN(_SUCCESS)
+   end function var_has_attr
+
+   function get_var_attr_real32(this,var_name,attr_name,rc) result(attr_real32)
+      class(FileMetadataUtils), intent(inout) :: this
+      character(len=*), intent(in) :: var_name
+      character(len=*), intent(in) :: attr_name
+      integer, optional, intent(out) :: rc
+
+      real(REAL32) :: attr_real32
```

Comment:
I prefer the declaration of the function result to be before the dummy arguments.   But only make a change if there is something else substantive in the review.

Reply:
I'm going to keep it things as they are rather than futz with checking for missing_value. I'll remember this in the future